### ISSUE TITLE
[mycelium] TYPE_CHECKING false-positive cycles: 0 confirmed after v0.1.10 upgrade

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#211]
## Summary

Mycelium v0.1.10 fixed the `if TYPE_CHECKING:` import-edge false-positive bug (#227 in
the mycelium repo). This issue documents the re-audit result for tree-sitter-analyzer.

## Before (≤ v0.1.9)

`mycelium detect-cycles --edge-kind imports` reported **7 cycle nodes** in
tree-sitter-analyzer. These were spurious: they traced imports inside
`if TYPE_CHECKING:` blocks (which are never executed at runtime).

## After (v0.1.10)

Re-running with v0.1.10:

```
$ mycelium detect-cycles --edge-kind imports --root .
{"count":0,"members":[]}
```

Zero import cycles confirmed. The codebase is genuinely acyclic at the runtime-import level.

## Call-graph improvements

v0.1.10 also fixed nested-attribute call tracking
(`self.history.append(x)` style). Re-index shows the call graph now has
**82,758 Calls edges** (up from ~80K with prior versions), indicating previously
lost edges have been recovered.

## Action

No code changes needed. Archive as reference for the next mycelium-driven audit.

**Comment by aimasteracc** [comment]
Mycelium v0.1.10 fixed the TYPE_CHECKING false-positive in its own issue #227. Re-audit on 2026-05-30 confirmed 0 cycle nodes remaining. This tracking issue is now obsolete.
